libs/community/langchain_community/vectorstores/azure_sql_server.py

```diff
@@ -0,0 +1,176 @@
+from typing import Any, Iterable, List, Optional
+
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from langchain_core.vectorstores import VST, VectorStore
+from sqlalchemy import Column, Uuid, bindparam, create_engine, text
+from sqlalchemy.dialects.mssql import JSON, NVARCHAR, VARBINARY
+from sqlalchemy.engine import Connection, Engine
+from sqlalchemy.exc import DBAPIError
+from sqlalchemy.orm import Session
+
+try:
+    from sqlalchemy.orm import declarative_base
+except ImportError:
+    from sqlalchemy.ext.declarative import declarative_base
+
+import json
+import logging
+import uuid
+
+Base = declarative_base()  # type: Any
+
+_embedding_store: Any = None
+
+
+class AzureSQLServer_VectorStore(VectorStore):
+    """Azure SQL Server Vector Store.
+
+    This class provides a vector store interface for adding texts and performing
+    similarity searches on the texts in Azure SQL Server.
+
+    """
+
+    def __init__(
+        self,
+        *,
+        connection: Optional[Connection] = None,
+        connection_string: str,
+        embedding_function: Embeddings,
+        table_name: str,
+    ) -> None:
+        """Initialize the Azure SQL Server vector store."""
+
+        self.connection_string = connection_string
+        self.embedding_function = embedding_function
+        self.table_name = table_name
+        self._bind = connection if connection else self._create_engine()
+        self.EmbeddingStore = self._get_embedding_store(table_name)
+        self._create_table_if_not_exists()
+
+    def _create_engine(self) -> Engine:
+        return create_engine(url=self.connection_string, echo=True)
+
+    def _create_table_if_not_exists(self) -> None:
+        logging.info("Creating table %s", self.table_name)
+        with Session(bind=self._bind) as session:
+            Base.metadata.create_all(session.get_bind())
+
+    def _get_embedding_store(self, name: str) -> Any:
+        global _embedding_store
+        if _embedding_store is not None:
+            return _embedding_store
+
+        class EmbeddingStore(Base):
+            """This is the base model for SQL vector store."""
+
+            __tablename__ = name
+            ID = Column(Uuid, primary_key=True, default=uuid.uuid4)
+            QUERYMETADATA = Column(JSON)
+            QUERY = Column(NVARCHAR)
+            VECTOR = Column(VARBINARY)
```

Comment:
I thought for V1, we are already looking to utilize the native type in latest SQL. So, shouldnt the type of vector NOT be varbinary?

Reply:
Yes, but the native data type isn't ready yet, even from the codebase (confirmed with Davide). He expects using VARBINARY for the time being.